### PR TITLE
Converted Drush commands to Drush 9.x compatible.

### PR DESCRIPTION
**Pull request by scottsawyer**
Adds Drush 9.x compatible commands for sync test and sync.

### CODE REVIEW

File: src/Commands/CommercePrintfulCommands.php [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      else {
+        $this->output()->writeln(dt('No Store Info found.'));
+      }
+    }
+    catch (PrintfulException $e) {
+      dt($e->getMessage);
+    }
+    try {
+      $products = $this->pf->syncProducts();
+      if ($products['code'] == 200 && isset($products['result'])) {
```
Comment (graber-1) [inline src/Commands/CommercePrintfulCommands.php:105]: It's not going to work as the Printful service throws an exception in such a case. Just do a try - catch block and catch a PrintfulException.

File: src/Commands/CommercePrintfulCommands.php [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+   */
+  public function syncProducts($bundle, $update) {
+
+    $batch = PrintfulSyncBatch::getBatch([
+      'product_bundle' => $bundle,
+      'update' => $update,
+    ]);
+    batch_set($batch);
+
+    drush_backend_batch_process();
```
Comment (graber-1) [inline src/Commands/CommercePrintfulCommands.php:136]: Yes, there's that drush function.

Maybe worth adding input validation and a try - catch block to catch a `PrintfulException`?

Just FYI, in VBO I did it like this (I don't remember why but had my reasons), using `drush_backend_batch_process` is OK though:

```    // Populate entity list.
    $context = [];
    do {
      $context['finished'] = 1;
      $context['message'] = '';
      ViewsBulkOperationsBatch::getList($vbo_data, $context);
      if (!empty($context['message'])) {
        $this->logger->info($context['message']);
      }
    } while ($context['finished'] < 1);
    $vbo_data = $context['results'];

    $this->timer($options['verbose'], 'list');

    // Execute the selected action.
    $context = [];
    do {
      $context['finished'] = 1;
      $context['message'] = '';
      ViewsBulkOperationsBatch::operation($vbo_data, $context);
      if (!empty($context['message'])) {
        $this->logger->info($context['message']);
      }
    } while ($context['finished'] < 1);
```

**Review by graber-1 — APPROVED** [review]
You can leave it as is and we can merge it if you don't have anymore time as it's new functionality anyway or we can make it a bit better according to comments. Up to you. Thanks!

File: src/Commands/CommercePrintfulCommands.php (hunk 2 above)
Comment (scottsawyer) [reply src/Commands/CommercePrintfulCommands.php:136]: Yeah, I did look at your VBO code.  I found that core removed the type hint for a bunch of stuff due to this very issue https://www.drupal.org/node/2664290.  I wonder if you ran into the same thing and just created your own work around?  Personal feeling is, if it's good enough for core, I am leaning toward leaving as-is.

File: src/Commands/CommercePrintfulCommands.php (hunk 1 above)
Comment (scottsawyer) [reply src/Commands/CommercePrintfulCommands.php:105]: Ah, that makes sense.

**Review by scottsawyer — COMMENTED** [review]
There are a lot of other things I'd like to do to improve the Drush commands, right now it just dumps the print_r into the output, so it's really ugly.  Would rather create a table or something.  There is also a command mentioned on the module page that I read about after I submitted the pull request (pgm), but didn't see an implementation in the 8.x version.  Honestly though, I feel like all these could be addressed later in a follow up.

**Pull request merged**